verification/types.go

```diff
@@ -120,13 +127,84 @@ var (
 	}
 )
 
-// FindVerificationLevel finds if the given string corresponds to a supported VerificationLevel, otherwise throws an error
-func FindVerificationLevel(s string) (*VerificationLevel, error) {
+// GetVerificationLevel finds if the given input corresponds to a valid VerificationLevel, otherwise throws an error
+func GetVerificationLevel(input interface{}) (*VerificationLevel, error) {
+	switch input.(type) {
+	case string:
+		for _, l := range VerificationLevels {
+			if l.Name == input {
+				return l, nil
+			}
+		}
+		return nil, fmt.Errorf("invalid signature verification %q", input)
+	case map[string]interface{}:
+		return parseCustomVerification(input)
+	}
+	return nil, fmt.Errorf("invalid signature verification %q", input)
+}
+
+func parseCustomVerification(input interface{}) (*VerificationLevel, error) {
+	cvl, ok := input.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("invalid custom signature verification")
+	}
+
+	var baseLevel *VerificationLevel
+	for _, l := range VerificationLevels {
+		if l.Name == cvl["level"] {
+			baseLevel = l
+		}
+	}
+	if baseLevel == nil {
+		return nil, fmt.Errorf("invalid custom signature verification %q", cvl)
+	}
+	if baseLevel == Skip {
+		return nil, fmt.Errorf("signature verification %q can't be used in custom signature verification", baseLevel.Name)
+	}
+
+	verificationLevel := &VerificationLevel{
+		Name:            "custom",
+		VerificationMap: make(map[VerificationType]VerificationAction),
+	}
+
+	// populate the custom verification level with the base verification settings
+	for k, v := range baseLevel.VerificationMap {
+		verificationLevel.VerificationMap[k] = v
+	}
+
+	// override the verification actions with the user configured settings
+	var overrideMap map[string]interface{}
+	if overrideMap, ok = cvl["override"].(map[string]interface{}); !ok {
+		return nil, fmt.Errorf("invalid custom signature verification %q", cvl)
+	}
+	for key, value := range overrideMap {
+		var verificationType VerificationType
+		for _, t := range VerificationTypes {
+			if strings.EqualFold(string(t), key) {
+				verificationType = t
+			}
+		}
+		if verificationType == "" {
+			return nil, fmt.Errorf("verification type %q in custom signature verification is not supported", verificationType)
```

Comment:
I feel `key` should be passed in instead of the empty `verificationType`

Reply:
Good catch. Addressed in the latest revision